django/db/models/expressions.py

```diff
@@ -126,6 +126,14 @@ class BaseExpression:
     is_summary = False
     _output_field = None
 
+    # if it can be used in a filter-clause
+    filterable = True
+
+    # flag to indicate if it's possible to use the expression in an update-query
+    # contains_aggregate will add a group by if contains_aggregate which may be
+    # incorrect for some expressions
+    can_be_reffed_in_update = True
```

Comment:
`for_save` is True for inserts and updates, so you can remove this and put the fielderror into the resolve_expressions that make sense.